### ISSUE TITLE
2.3.x introduced false error TS2449: Class used before its declaration

### ISSUE THREAD

**Issue by evil-shrike** [Microsoft/TypeScript#15786]
**TypeScript Version:**  2.3.1 , 2.3.2

**Code**

```ts
import LocalState = DomainObject.LocalState;
//                             ^^^^^^^^^ error TS2449: Class 'DomainObject' used before its declaration.
class DomainObject {
	localStates: typeof LocalState;
}
namespace DomainObject {
	export enum LocalState {}
}
```

**Expected behavior:**
No error.

**Actual behavior:**
error TS2449: Class 'DomainObject' used before its declaration.

**Comment by evil-shrike** [comment]
Probably it's related to #14679

**Comment by mhegazy** [comment]
This is actually behaving as intended. `import` are aliases, they can be used in both value and type positions. it is unsafe to use them in value, since you are guaranteed here that `LocalState` is `undefined`.

If all what you need is a type alias, then consider using `type` instead:
```ts
type LocalState = typeof DomainObject.LocalState;
class DomainObject {
	localStates: LocalState ;
}
```

**Comment by evil-shrike** [comment]
it's an interface, it cannot be a value.

also I can't see any difference between:
1. not works:
```ts
import LocalState = DomainObject.LocalState;
class DomainObject {
	localStates: typeof LocalState ;
}
```
2. works:
```ts
class DomainObject {
	localStates: typeof DomainObject.LocalState ;
}
```

**Comment by evil-shrike** [comment]
(changed comment)
Ok. But it could work in cases when it's used as type (as in the example). It looks a bit overstrict.

Also I have different cases:
```
const DefaultMapping: PropertyEditorMap = {
	peNotAuthorized: PropertyEditor,
//                                  ^^^^^ error 
};

class PropertyEditor extends Component {
}
```

**Comment by mhegazy** [comment]
> Ok. But it could work in cases when it's used as type (as in the example). It looks a bit overstrict.

that is correct, but the compiler does not know for sure it will not be used as a value. 

the second issue is a bug.

**Comment by evil-shrike** [comment]
Actually the second case is the same. Class is used as value and its not defined yet. So it's correct check. Closing.
